python/packages/ai/teams/ai/data_sources/data_source.py

```diff
@@ -17,8 +17,10 @@ class DataSource(ABC):
     A data source that can be used to render text that's added to a prompt.
     """
 
-    name: str
-    "Name of the data source."
+    @property
+    @abstractmethod
```

Comment:
Is `@abstractmethod` equivalent to `return self._name`?

Reply:
Marking it as `@abstractmethod` lets people determine their own logic to provide the name. For example, they may want to use another variable to store the name, or calculate the name instead of just returning a private property.
Removing `@abstractmethod` and let it `return self._name` you proposed also works, we need to have a constructor in the abstract class to ensure the property is initialized in this case.

Let me know which way you prefer.

Reply:
I think this makes sense, @lilyydu this is essentially in typescript a getter with a default implementation that can be overridden by a class that extends it.

```typescript
export class DataSource {
    get name(): string {
        return "name";
    }
}

export class MyDataSource extends DataSource {
    get name(): string { return this._name; }
    private _name: string;
}
```

Reply:
Gotcha, this works fine by me- thank you both for explaining!